src/main/java/com/spotify/dns/AggregatingChangeNotifier.java

```diff
@@ -29,7 +29,7 @@
 
   private final List<ChangeNotifier<T>> changeNotifiers;
 
-  private volatile Set<T> records = ChangeNotifiers.initialEmptyDataInstance();
+  private volatile Set<T> records;
```

Comment:
Doesn't this change the initial notification logic?

Reply:
I mean, it seems like either, it would stop working, or it would no longer need the `initialEmptyDataInstance()` method and the associated constant?

Reply:
IntelliJ is saying that the field is redundantly set since it's also set in the constructor. But we're doing some work in the constructor which might led to it being read. Will look through that call tree once more and see what's actually happening.

Reply:
On further inspection, it looks like the initial value is actually never read and can be removed safely. The `initialEmptyDataInstance` constant is used in other implementations of `ChangeListener` so I can't remove it.

Reply:
I agree, I had forgotten that there were other kinds of change notifiers.